File: api/models/Movie.go
package models

import (
	"time"

	"gorm.io/gorm"
)

type Movie struct {
	gorm.Model
	Title           string        `gorm:"size:80;not null;index" json:"title"`
	TmdbID          string        `gorm:"not null;default:'unknown';index" json:"tmdbID"`
	ImdbID          string        `gorm:"not null;default:'unknown';index" json:"imdbID"`
	Thumb           string        `gorm:"default:'/public/images/no_thumb.jpg'" json:"thumb"`
	Art             string        `gorm:"default:'/public/images/no_art.jpg'" json:"art"`
	Tagline         string        `gorm:"default:'unknown'" json:"tagline"`
	Summary         string        `gorm:"type:text" json:"summary"`
	Budget          uint          `json:"budget"`
	Revenue         uint          `json:"revenue"`
	ReleaseDate     time.Time     `json:"releaseDate"`
	Year            uint          `json:"year"`
	ContentRating   string        `gorm:"size:20" json:"contentRating"`
	AudienceRating  float32       `gorm:"default:0.0" json:"audienceRating"`
	CriticRating    float32       `gorm:"default:0.0" json:"criticRating"`
	Duration        uint          `json:"duration"`
	SpokenLanguages string        `gorm:"default:'unknown'" json:"spokenLanguages"`
	Resolution      string        `gorm:"size:12" json:"resolution"`
	Artists         []*Artist     `gorm:"many2many:artist_movies" json:"artists"`
	Studios         []*Studio     `gorm:"many2many:movie_studios" json:"studios"`
	Genres          []*MovieGenre `gorm:"many2many:movie_genre" json:"genres"`
	Users           []*User       `gorm:"many2many:user_movie" json:"users"`
}

File: tv/types/Movie.ts
import type { Genre } from "./Genre";
import type { Studio } from "./Studio";
import type { Crew } from "./Crew";
import type { Cast } from "./Cast";
import type { Video } from "./Video";
import type { Audio } from "./Audio";
import type { Subtitles } from "./Subtitles";
import type { MovieExtras } from "./MovieExtras";

export type SimpleMovie = {
  ID: number;
  title: string;
  thumb: string;
  year: number;
};

export type MovieCardProps = {
  movie: SimpleMovie;
  hasTVPreferredFocus?: boolean;
};

export type MoviesResponse = {
  movies: SimpleMovie[];
};

export type MovieResponse = {
  movie: Movie;
};

export type Movie = {
  ID: number;
  title: string;
  filePath: string;
  FileName: string;
  container: string;
  size: number;
  contentType: string;
  resolution: number;
  runTime: number;
  tagLine: string;
  summary: string;
  thumb: string;
  art: string;
  tmdbID: string;
  imdbID: string;
  releaseDate: string | Date;
  year: number;
  budget: number;
  revenue: number;
  contentRating: string;
  audienceRating: number;
  criticRating: number;
  spokenLanguages: string;
  genres: Genre[];
  studios: Studio[];
  castList: Cast[];
  crewList: Crew[];
  videoList: Video[];
  audioList: Audio[];
  subtitleList: Subtitles[];
  extras: MovieExtras[];
  CreatedAt?: string | Date;
  UpdatedAt?: string | Date;
  DeletedAt?: string | Date | null;
};
